### ISSUE TITLE
Drawing svg during scrolling

### ISSUE THREAD

**Issue by rotellanomala** [maxwellito/vivus#211]
Dear Maxwellito,
first of all, thanks for your script and sorry for my bad English,
I'm an Italian graphic designer, not a developer, so I ask you sorry also for what I will to submit.

I'm working with your library for a webpage, playing some svgs, in different ways, timings and delays.

After playing successfully the first ones, I would like to draw an svg during scrolling.

The issue is that works locally only with Safari,
nothing happens with Firefox and Chrome,
while it does not work on the network with any browser,

I'm quite sure the problem is my script, I'm not a developer
and I would like to ask you an help.

Here's the link to a simplified part of my project.
https://jsfiddle.net/rotella/Lfqrzpod/

Thanks in advance for your help,
alberto.

**Comment by maxwellito** [comment]
Hey there,

Sorry but I got a pretty busy weekend. I think I found a temporary solution
to your problem. I'll come back with some details later tomorrow
https://jsfiddle.net/r9wyxofv/

Le sam. 8 déc. 2018 à 13:07, rotellanomala <notifications@github.com> a
écrit :

**Comment by rotellanomala** [comment]
Dear Maxwellito,
thanks a lot for your fast reply and "temporary" solution,
for me works just great!
I've also tried with _progress_,
something like this

```
var drawLogo = new Vivus ('logoLettering', {type: 'sync', start:'manual'});
				
				function resetta(){
					drawLogo.reset();
				}
				
		window.addEventListener('scroll', scrolling);		
				
				function scrolling(){
					
					if (document.body.scrollTop > 0 || document.documentElement.scrollTop > 0){
					
						progress = window.scrollY/100;
						console.log(progress);
						drawLogo.setFrameProgress(progress);
						drawLogo.stop();
						
					}
					
				}
```


but having same type of issues.

So your solution is perfect!

Don't worry and take your time, I'm not in a hurry.

Thanks again and have a nice weekend.
alberto

**Comment by maxwellito** [comment]
Brilliant then :) You were pretty close, you were listening on the wrong
event. The event 'scroll' is triggered when the target is scrolled, but the
body of your page is not big enough for scrolling. But the event 'wheel' is
used when the user use the wheel on the target.
Enjoy then :)

If you have more questions, don't hesitate

**Comment by maxwellito** [comment]
The CSS change I did was something I forgot to remove. It's not even valid. You can get rid of it. Only the event type is important :)
So, all good now?

**Comment by rotellanomala** [comment]
Hi there,
sorry for my delay,
regarding the CSS changes, I've noticed that

with the _**#testLogo #logo**_ position property (of the [jsFiddle sample](https://jsfiddle.net/r9wyxofv/)) set to "absolute", doesn't work,
while with the "relative" position works great,
so I thought that, besides the event type, was also important the positioning.

Anyway, as I said, I'm not a developer, so I would not say nonsense,

now I'm working with your solution and it's ok!

Thanks a lot for your valuable help!!!
Thanks for your awesome work!!!

Have a nice day,
my best regards,
alberto.

**Comment by rotellanomala** [comment]
Dear Maxwellito,

sorry if I write a new comment,

I would only like to thank you once again
and let you know that your solution with deltay wheel event works great!

I've just finished redoing my page, implementing your precious code,
at first, it was fine for Firefox and Chrome but not in Safari :confounded:,
the opposite status of my original issue,
even more clear then that it was my code full of bugs!

I've found that, within my page, _deltay wheel event_ wasn't work properly on Safari due to my html/css "confusion", by adding a simply _overflow:hidden_ property to the body all works great!

So, thanks for your great work and precious help!

Happy holidays,
alberto.
